sunpy/net/dataretriever/sources/goes.py

```diff
@@ -168,9 +178,95 @@ def _can_handle_query(cls, *query):
         boolean
             answer as to whether client can service the query
         """
-        chkattr = ['Time', 'Instrument', 'SatelliteNumber']
+        chkattr = ["Time", "Instrument", "SatelliteNumber"]
         chklist = [x.__class__.__name__ in chkattr for x in query]
         for x in query:
-            if x.__class__.__name__ == 'Instrument' and x.value.lower() in ('xrs', 'goes'):
+            if x.__class__.__name__ == "Instrument" and x.value.lower() in (
+                "xrs",
+                "goes",
+            ):
                 return all(chklist)
         return False
+
+
+class SUVIClient(GenericClient):
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        Returns a URL to the SUVI data for the specified date.
+
+        Parameters
+        ----------
+        timerange: sunpy.time.TimeRange
+            time range for which data is to be downloaded.
+
+        level : str
+            The level of the data.
+
+        Note
+        ----
+        SUVI data does not exist for GOES satellite < 16.
+        """
+        base_url = "https://data.ngdc.noaa.gov/platforms/solar-space-observing-satellites/goes/goes{goes_number}/"
+
+        if "wavelength" not in kwargs.keys() or not kwargs["wavelength"]:
+            raise ValueError("Queries must specify a wavelength.")
+        else:
+            wavelength = kwargs["wavelength"]
+        wavelength = wavelength.to(u.Angstrom, equivalencies=u.spectral())
+
+        if "satellitenumber" in kwargs.keys():
+            satellitenumber = int(kwargs["satellitenumber"])
+        else:
+            satellitenumber = 16
+
+        if "level" in kwargs.keys():
```

Comment:
and this one should be:
```suggestion
level = str(kwargs.get("level", "2"))
```